### PR TITLE
Kraken: Implement create vj in mvj

### PR DESCRIPTION
**Pull request by xlqian**

### CODE REVIEW

File: source/ed/build_helper.cpp [hunk 1]
```diff
@@ -64,14 +64,28 @@ VJ::VJ(builder & b, const std::string &line_name, const std::string &validity_pa
...
+        auto idx = pt_data.vehicle_journeys.size();
+        name = "vehicle_journey " + std::to_string(idx);
+    }
+
     if (is_frequency) {
-        auto f_vj = std::make_unique<nt::FrequencyVehicleJourney>();
-        vj = f_vj.get();
-        route->frequency_vehicle_journey_list.push_back(std::move(f_vj));
+        // NOTE: the meta vj name should be the same as the vj's name
+        mvj = pt_data.meta_vjs.get_or_create(name);
```
Comment (antoine-de) [inline source/ed/build_helper.cpp]: you can move this line above no ?

File: source/type/type.h [hunk 2]
```diff
@@ -1021,9 +1019,11 @@ struct MetaVehicleJourney: public Header, HasMessages {
     get_vj_at_date(RTLevel level, const boost::gregorian::date& date) const;
     std::vector<VehicleJourney*>
     get_vjs_in_period(RTLevel level, const boost::gregorian::date_period& period) const;
-
 private:
-    navitia::flat_enum_map<RTLevel, std::vector<VehicleJourney*>> rtlevel_to_vjs_map;
+    template<typename VJ>
+    VJ* set_vj(std::unique_ptr<VJ>& vj_ptr, RTLevel level);
+private:
```
Comment (antoine-de) [inline source/type/type.h]: useless `private`

Comment (xlqian) [inline source/type/type.h]: I'm used to have a private section for members and a private section for functions... but ok, i'll remove it

**Comment by xlqian** [comment]
Jenkins, test this please

**Comment by pbougue** [comment]
Jenkins, test this please

**Comment by pbougue** [comment]
If I'm not mistaken, you can add "Kraken:" to the title

File: source/ed/ed_reader.cpp [hunk 3]
```diff
@@ -855,40 +869,13 @@ void EdReader::fill_associated_calendar(nt::Data& data, pqxx::work& work) {
...
-        const std::string vj_class = const_it["vj_class"].as<std::string>();
-        auto level = navitia::type::get_rt_level_from_string(vj_class);
-        meta_vj->add_vj(vj, level);
-        vj->meta_vj = meta_vj;
-        data.pt_data->headsign_handler.change_name_and_register_as_headsign(*vj, vj->name);
-    }
-
     //then we fill the links between a metavj and its associated calendars
-    request = "SELECT l.name as name, meta.associated_calendar_id as associated_calendar_id,"
+    std::string request = "SELECT l.name as name, meta.associated_calendar_id as associated_calendar_id,"
```
Comment (TeXitoi) [inline source/ed/ed_reader.cpp]: you can const it.

**Comment by TeXitoi** [comment]
OK for me.

File: source/type/data.cpp [hunk 4]
```diff
@@ -341,7 +341,7 @@ void Data::complete(){
 
 static ValidityPattern get_union_validity_pattern(const MetaVehicleJourney& meta_vj) {
     ValidityPattern validity;
-    for (const auto* vj: meta_vj.get_base_vj()) {
+    for (const auto& vj: meta_vj.get_base_vj()) {
```
Comment (pbougue) [inline source/type/data.cpp:344]: :+1:

**Pull request merged**